src/main/java/net/sf/jabref/logic/importer/fileformat/MrDLibImporter.java

```diff
@@ -0,0 +1,275 @@
+                        authors = true;
+                    }
+                    if (qName.equalsIgnoreCase("published_in")) {
+                        published_in = true;
+                    }
+                    if (qName.equalsIgnoreCase("title")) {
+                        title = true;
+                    }
+                    if (qName.equalsIgnoreCase("year")) {
+                        year = true;
+                    }
+                    if (qName.equalsIgnoreCase("type")) {
+                        type = true;
+                    }
+                    //-----------
+                    if (qName.equalsIgnoreCase("related_article")) {
+                        htmlSnippetSingle = null;
+                        htmlSnippetSingleRank = -1;
+                    }
+                    if (qName.equalsIgnoreCase("snippet")
+                            && attributes.getValue(0).equalsIgnoreCase("html_fully_formatted")) {
+                        snippet = true;
+                    }
+                    if (qName.equalsIgnoreCase("suggested_rank")) {
+                        rank = true;
+                    }
+
+                }
+
+                @Override
+                public void endElement(String uri, String localName, String qName) throws SAXException {
+                    if (qName.equalsIgnoreCase("related_article")) {
+                        rankedBibEntries.add(new RankedBibEntry(currentEntry, htmlSnippetSingleRank));
+                        currentEntry = new BibEntry();
+                    }
+                }
+
+                @Override
+                public void characters(char ch[], int start, int length) throws SAXException {
+
+                    if (authors) {
+                        currentEntry.setField("author", new String(ch, start, length));
+                        authors = false;
+                    }
+                    if (published_in) {
+                        currentEntry.setField("journal", new String(ch, start, length));
+                        published_in = false;
+                    }
+                    if (title) {
+                        currentEntry.setField("title", new String(ch, start, length));
+                        title = false;
+                    }
+                    if (year) {
+                        currentEntry.setField("year", new String(ch, start, length));
+                        year = false;
+                    }
+                    // -------------
+                    if (rank) {
+                        htmlSnippetSingleRank = Integer.parseInt(new String(ch, start, length));
+                        rank = false;
+                    }
+                    if (snippet) {
+                        currentEntry.setField("html_representation", new String(ch, start, length));
+                        snippet = false;
+                    }
+
+                }
+
+            };
+
+            try {
```

Comment:
see as above, use the Autocloseable feature try ( InputStream strem = ....) , it takes care of auto closing the stream, even in the exception case